internal/eval/deque.go

```diff
@@ -215,6 +302,179 @@ func (q *Deque) RPop() (string, error) {
 	return x, nil
 }
 
+func (q *Deque) breakPivotNodeAndInsertAfter(qIterator *DequeIterator, pivotNode *byteListNode, element string) *byteListNode {
+	newNode := q.list.newNode()
+	if pivotNode.next != nil {
+		pivotNode.next.prev = newNode
+	}
+	newNode.next = pivotNode.next
+	pivotNode.next = nil
+	newNode.buf = append([]byte{}, pivotNode.buf[qIterator.BufIndex:]...)
+	pivotNode.buf = pivotNode.buf[:qIterator.BufIndex]
+	q.list.tail = pivotNode
+	q.RPush(element)
+	newNode.prev = q.list.tail
+	q.list.tail.next = newNode
+	return newNode
+}
+
+func (q *Deque) insertAfterPivotNodeHelper(element string, qIterator *DequeIterator, pivotNode *byteListNode) {
+	prevTail := q.list.tail
+	if qIterator.BufIndex == 0 {
+		pivotNode.next = nil
+		q.list.tail = pivotNode
+		q.RPush(element)
+		q.list.tail.next = qIterator.CurrentNode
+		qIterator.CurrentNode.prev = q.list.tail
+		q.list.tail = prevTail
+	} else {
+		newNode := q.breakPivotNodeAndInsertAfter(qIterator, pivotNode, element)
+		if newNode.next == nil {
+			q.list.tail = newNode
+		} else {
+			q.list.tail = prevTail
+		}
+	}
+}
+
+func (q *Deque) insertAfterPivotNode(element string, qIterator *DequeIterator, pivotNode *byteListNode) {
+	if qIterator.ElementsTraversed == q.Length {
+		// Element needs to be inserted at the end of the Deque.
+		q.RPush(element)
+	} else {
+		// Element needs to be inserted b/w 2 nodes in the Deque.
+		q.insertAfterPivotNodeHelper(element, qIterator, pivotNode)
+	}
+}
+
+func (q *Deque) getNewNodeWithElement(element string) *byteListNode {
+	elementEntryLen := int(GetEncodeDeqEntrySize(element))
+	elementNode := q.list.newNodeWithCapacity(elementEntryLen)
+	elementNode.buf = elementNode.buf[:elementEntryLen]
+	EncodeDeqEntryInPlace(element, elementNode.buf[:elementEntryLen])
+	return elementNode
+}
+
+func (q *Deque) breakPivotNodeAndInsertBefore(qIterator *DequeIterator, pivotNode, elementNode *byteListNode, pivotEntryLen, leftIdx int) *byteListNode {
```

Comment:
Add a comment explaining what this method does.

Reply:
Done.